json-logic/questionnaire.ts

```diff
@@ -0,0 +1,103 @@
+// This file showcases a basic reference implementation.
+
+import { IQuestion, QuestionType, IQuestionnaire, IQuestionnaireMeta, IVariable } from "./schema"
+import { LogicConstant, LogicExpression, jsonLogic } from "./logic";
+
+declare var jsonLogic: jsonLogic;
+
+abstract class Question implements IQuestion {
+	id: string;
+	type: QuestionType;
+	text: string;
+	skipIf?: LogicExpression;
+
+	abstract ask() : LogicConstant;
+
+	public check(state: IQuestionnaire) : boolean {
+		if(this.skipIf) {
+			return !jsonLogic.apply(this.skipIf, state.variables);
+		}
+		else {
+			return true;
+		}
+	}
+}
+
+
+
+class Questionnaire implements IQuestionnaire {
+	meta: IQuestionnaireMeta;
+	questions: Question[];
+	variables: IVariable[];
+	answeredQuestions: string[] = [];
+	resultCategories: [];
+
+	public nextQuestion() {
+		// start always from first question, because the guard might evaluate to true
```

Comment:
I would not allow a jump backwards in the array of questions. I would just iterate over the array and show each question if its condition is fulfilled.